scrape/manager_test.go

```diff
@@ -616,6 +616,42 @@ func TestManagerTargetsUpdates(t *testing.T) {
 	}
 }
 
+func TestManagerDuplicateAfterRelabellingWarning(t *testing.T) {
+	var buf bytes.Buffer
+	writer := &buf
+	logger := promslog.New(&promslog.Config{Writer: writer})
+
+	opts := Options{WarnDuplicateTargets: true}
+	testRegistry := prometheus.NewRegistry()
+	m, err := NewManager(&opts, logger, nil, nil, testRegistry)
+	require.NoError(t, err)
+
+	m.scrapePools = map[string]*scrapePool{}
+	sp := &scrapePool{
+		activeTargets: map[uint64]*Target{},
+	}
+	targetScrapeCfg := &config.ScrapeConfig{
+		Scheme:         "https",
+		MetricsPath:    "/metrics",
+		JobName:        "job",
+		ScrapeInterval: model.Duration(time.Second),
+		ScrapeTimeout:  model.Duration(time.Second),
+	}
+
+	sp.activeTargets[uint64(0)] = &Target{
```

Comment:
I feel this is too low‑level and too tightly coupled to the implementation, how about passing a config like in https://github.com/prometheus/prometheus/blob/091e662f4d2e7b0e0310844fe49d7297d17f1bc4/scrape/scrape_test.go#L5173-L5204, trigger and reload and wait for the warning?